docs: update `expect.soft` related documentation (#3535)
vitest-dev/vitest: docs/api/expect.md

`expect` has no effect on testing types, if the expression doesn't have a type error. If you want to use Vitest as [type checker](/guide/testing-types), use [`expectTypeOf`](/api/expect-typeof) or [`assertType`](/api/assert-type).
:::

## soft

- **Type:** `ExpectStatic & (actual: any) => Assertions`

>   `expect.soft` is the same as `expect` but does not terminate test execution when the assertion fails. It will be marked as a test failure. Until the test ends, all errors will be output.

Reviewer: ```suggestion
`expect.soft` functions similarly to `expect`, but instead of terminating the test execution upon a failed assertion, it continues running and marks the failure as a test failure. All errors encountered during the test will be displayed until the test is completed.
```

Author: Thank you for the review!

  ```ts
  import { expect } from 'vitest'

  test('expect.soft test', () => {
    expect.soft(1 + 1).toBe(3) // mark the test as fail and continue
    expect.soft(1 + 2).toBe(4) // mark the test as fail and continue
  })
  // At the end of the test, the above errors will be output.
  ```
  

>   it can also be used with `expect`. if `expect` assertion fails, the test will be terminate and all errors will be output.

Reviewer: ```suggestion
  it can also be used with `expect`. if `expect` assertion fails, the test will be terminated and all errors will be displayed.
```

  ```ts
  import { expect } from 'vitest'

  test('expect.soft test', () => {
    expect.soft(1 + 1).toBe(3) // mark the test as fail and continue
    expect(1 + 2).toBe(3) // failed and terminate the test, all previous errors will be output
    expect.soft(1 + 2).toBe(4) // do not run
  })
  ```
  

>   > **Note:** `expect.soft` only supported run inside the [test](/api/#test) function.

Reviewer 2: We use `::: note` syntax for notes and warnings

And this looks more like a warning to me

`expect` has no effect on testing types, if the expression doesn't have a type error. If you want to use Vitest as [type checker](/guide/testing-types), use [`expectTypeOf`](/api/expect-typeof) or [`assertType`](/api/assert-type).
:::

## soft

- **Type:** `ExpectStatic & (actual: any) => Assertions`

`expect.soft` functions similarly to `expect`, but instead of terminating the test execution upon a failed assertion, it continues running and marks the failure as a test failure. All errors encountered during the test will be displayed until the test is completed.

  ```ts

>   import { expect } from 'vitest'

Reviewer 2: ```suggestion
  import { expect, test } from 'vitest'
```

  test('expect.soft test', () => {
    expect.soft(1 + 1).toBe(3) // mark the test as fail and continue
    expect.soft(1 + 2).toBe(4) // mark the test as fail and continue
  })
  // At the end of the test, the above errors will be output.
  ```
  

>   it can also be used with `expect`. if `expect` assertion fails, the test will be terminated and all errors will be displayed.

Reviewer 2: ```suggestion
  It can also be used with `expect`. if `expect` assertion fails, the test will be terminated and all errors will be displayed.
```

  ```ts

>   import { expect } from 'vitest'

Reviewer 2: ```suggestion
  import { expect, test } from 'vitest'
```

  test('expect.soft test', () => {
    expect.soft(1 + 1).toBe(3) // mark the test as fail and continue
    expect(1 + 2).toBe(3) // failed and terminate the test, all previous errors will be output
    expect.soft(1 + 2).toBe(4) // do not run
  })
  ```
  
::: warning

> `expect.soft` only supported run inside the [test](/api/#test) function.

Reviewer 2: ```suggestion
`expect.soft` can only be used inside the [`test`](/api/#test) function.
```
